Sources/TuistSupport/Extensions/String+Extras.swift

```diff
@@ -4,7 +4,8 @@ import struct TSCUtility.Version
 extension String {
     // swiftlint:disable:next force_try
     private static let whitespaceRegularExpression = try! NSRegularExpression(pattern: "\\s")
-    static let httpRegularExpression = "(\\w+://)(.+@)*([\\w\\d\\.]+)(:[\\d]+){0,1}/*(.*)"
+    // Also possible suggestion from ChatGPT: (?!ssh:\\/\\/)(\\w+://)(.+@)*([\\w\\d\\.]+)(:[\\d]+){0,1}/*(.*)
+    static let httpRegularExpression = "(http(s)?)(://)(.+@)*([\\w\\d\\.]+)(:[\\d]+){0,1}/*(.*)"
```

Comment:
is this fixing a problem? which one?

Reply:
It fixes the 3rd point in my initial PR post: Git urls containing user names wasn't matched correctly.

Example:

`ssh://git@some.server.com:7999/proejcts/my-templates.git@feature/adapt-simplifier`

Because the original regex was supporting git urls with branch names that is separated by and `@` at the end. But user names are in the front and would then contain 2x `@` signs in an url